pipcompilemulti/features/controller.py

```diff
@@ -61,10 +64,11 @@ def save_command_options(*args, **kwargs):
 
     def pin_options(self, env_name):
         """Return list of options to pin command."""
-        options = self.use_cache.pin_options()
-        options.extend(self.add_hashes.pin_options(env_name))
-        options.extend(self.allow_unsafe.pin_options())
-        return options
+        options = set(self.use_cache.pin_options())
+        options |= set(self.add_hashes.pin_options(env_name))
+        options |= set(self.allow_unsafe.pin_options())
+        options |= set(self.index.pin_options())
+        return list(options)
```

Comment:
It's better to have options as a list so that features could use positional arguments.
E.g. `['--upgrade-only', 'click']`.